src/thread.cpp

```diff
@@ -118,7 +118,7 @@ namespace libtorrent
 		TORRENT_ASSERT(l.locked());
 		++m_num_waiters;
 		l.unlock();
-		WaitForSingleObject(m_sem, INFINITE);
+		WaitForSingleObjectEx(m_sem, INFINITE, FALSE);
```

Comment:
the FALSE here is an optimization I take it, right? To prevent spurious wakeups.